app/store/validateMigration/validateMigration.ts

```diff
@@ -0,0 +1,36 @@
+import Logger from '../../util/Logger';
+import { RootState } from '../../reducers';
+import { ValidationCheck } from './validateMigration.types';
+
+// checks
+import { validateAccountsController } from './accountsController';
+import { validateKeyringController } from './keyringController';
+import { validateEngineInitialized } from './engineBackgroundState';
+
+const checks: ValidationCheck[] = [
+  validateEngineInitialized,
+  validateAccountsController,
+  validateKeyringController,
+];
+
+/**
+ * Runs all validations and logs any errors, but doesn’t throw.
+ * This makes sure your app keeps running even if some data is unexpected.
+ */
+export function validatePostMigrationState(state: RootState): void {
+  const allErrors: string[] = [];
+
+  for (const check of checks) {
+    const errors = check(state);
+    if (errors.length > 0) {
+      allErrors.push(...errors);
+    }
+  }
```

Comment:
You can simplify it like this

```suggestion
  const allErrors = checks.flatMap(check => check(state));
```

Reply:
good call.